### PR TITLE
dbr quant overhead[12/x]: turn off overrides for module convert output hook

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #69002 [not for land] dbr quant: turn off overrides on regions of model
* #69001 [not for land] dbr quant: add any_child_needs_op_hooks flag
* #69000 [not for land] dbr quant: add any_child_needs_arg_dequants flag
* #68999 [not for lnad] dbr quant: improve needs_dtype_transform_on_outputs
* #68877 dbr quant overhead [14/x]: cache whether an op is a module
* #68841 dbr quant overhead [13/x]: cache results of get_module_hook_type
* #68840 dbr quant: fix debugging fqn info for converted model
* **#68839 dbr quant overhead[12/x]: turn off overrides for module convert output hook**
* #68838 dbr quant: remove unnecessary outputs hook
* #68837 dbr quant overhead[11/x]: speed up module convert hook
* #68836 dbr quant overhead[10/x]: disable torch_function overrides for leaf nodes

Summary:

We can assume that there are no overrides needed for the hook which
dequantizes the module outputs, so we can turn them off explicitly.
While this does not lead to a measurable perf win, it makes things
easier to debug by eliminating the no-op overrides.

Test plan:

```
python test/test_quantization.py TestQuantizeDBR
```

Differential Revision: [D32630886](https://our.internmc.facebook.com/intern/diff/D32630886)

### CODE REVIEW

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32630886).

**Comment by vkuzo** [comment]
@vkuzo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32630886).

File: torch/ao/quantization/_dbr/auto_trace.py [hunk 1]
```diff
@@ -512,7 +512,18 @@ def _patched_module_call(self, *args, **kwargs):
...
+                        # For the sake of performance, we assume no overrides
+                        # are needed for quantizing/dequantizing things
+                        old_global_disable_torch_function_override = \
+                            global_disable_torch_function_override
+                        global_disable_torch_function_override = True
+
                         output = cur_qstate.outputs_convert_hook(output)
+
+                        global_disable_torch_function_override = \
+                            old_global_disable_torch_function_override
```
Comment (jerryzh168) [inline torch/ao/quantization/_dbr/auto_trace.py:525]: wondering if we can use with for this pattern

**Review by jerryzh168 — APPROVED** [review]

**Review by t10-13rocket — COMMENTED** [review]
LGTM

**Pull request closed without merging**